### PR TITLE
Kernel ideals and universal property of ring quotients

### PR DESCRIPTION
**Pull request by felixwellen**
Added RingHoms, kernels of RingHoms as ideals and the universal property of ring quotients.
The history might be confusing, since algebra code was restructered in between. 
The only real updates from this PR are Kernel.agda, some extension of QuotientRing.agda, a new record 'RingHom' in Ring.Base and some minor changes in Ring.Properties.

### CODE REVIEW

**Comment by felixwellen** [comment]
No, actually it is also confusing because I messed it up...

**Comment by felixwellen** [comment]
No the changes fit what I wrote.

File: Cubical/Algebra/Ring/Properties.agda
Comment (mortberg) [inline Cubical/Algebra/Ring/Properties.agda:34]: Why do this change? I kind of prefer to omit the angle brackets around R...

File: Cubical/Algebra/Kernel.agda
Comment (mortberg) [inline Cubical/Algebra/Kernel.agda:2]: Calling this file Kernel is maybe not ideal (pun intended!) as it's just about kernels in rings... Maybe we should have a subfolder Algebra.Ring where we have definitions like this? Alternatively calling thing `RingKernel` or something, but I prefer adding a subfolder

**Review by mortberg — CHANGES_REQUESTED** [review]
Looks mostly great to me! But what happened to "Cubical/Algebra/Monoid/Base.agda"? Github claims that it has been completely changed, but I can't see any difference to the actual code...

File: Cubical/Algebra/Kernel.agda
Comment (felixwellen) [reply Cubical/Algebra/Kernel.agda:2]: I didn't think about it, I like the subfolder idea.

**Comment by felixwellen** [comment]
> 
> 
> Looks mostly great to me! But what happened to "Cubical/Algebra/Monoid/Base.agda"? Github claims that it has been completely changed, but I can't see any difference to the actual code...

I'll try to figure out what happened. I missed it when looking at the changes because it was collapsed.

File: Cubical/Algebra/Ring/Properties.agda
Comment (felixwellen) [reply Cubical/Algebra/Ring/Properties.agda:34]: I'll revert that - I don't have a strong preference and I don't remember why I changed that.

File: Cubical/Algebra/Kernel.agda
Comment (felixwellen) [reply Cubical/Algebra/Kernel.agda:2]: I'll also move Ideal and QuotientRing to Ring/

File: Cubical/Algebra/Ring/Properties.agda
Comment (mortberg) [reply Cubical/Algebra/Ring/Properties.agda:34]: This should get better once we go for partially bundled structures as suggested by: https://github.com/agda/cubical/issues/405

Comment (felixwellen) [reply Cubical/Algebra/Ring/Properties.agda:34]: Good!

**Review by mortberg — APPROVED** [review]

**Comment by mortberg** [comment]
Perfect! Will merge once Travis is finished.

@mzeuner : I guess Localization should go in the Algebra.Ring subfolder once I've merged this

**Pull request merged**
